src/librustdoc/html/render/mod.rs

```diff
@@ -110,6 +110,11 @@ crate struct Context<'tcx> {
     /// real location of an item. This is used to allow external links to
     /// publicly reused items to redirect to the right location.
     render_redirect_pages: bool,
+    /// The map used to ensure all generated 'id=' attributes are unique.
+    id_map: RefCell<IdMap>,
+    /// Tracks section IDs for `Deref` targets so they match in both the main
+    /// body and the sidebar.
+    deref_id_map: RefCell<FxHashMap<DefId, String>>,
```

Comment:
I wasn't able to get rid of the RefCells here because the fields are used all over the place.